File: app/src/main/java/de/westnordost/streetcomplete/data/osm/changes/StringMapEntryChange.java
package de.westnordost.streetcomplete.data.osm.changes;

import java.util.Map;

public interface StringMapEntryChange
{
	void applyTo(Map<String,String> map);
	String toString();
	boolean conflictsWith(Map<String,String> map);
	boolean equals(Object other);
	StringMapEntryChange reversed();
}

File: app/src/main/java/de/westnordost/streetcomplete/data/osm/changes/StringMapEntryChange.kt
package de.westnordost.streetcomplete.data.osm.changes

interface StringMapEntryChange {
    override fun toString(): String
    override fun equals(other: Any?): Boolean
    fun conflictsWith(map: Map<String, String>): Boolean
    fun applyTo(map: MutableMap<String, String>)
    fun reversed(): StringMapEntryChange
}
